File: app/src/main/java/forpdateam/ru/forpda/client/OkHttpResponseException.java
package forpdateam.ru.forpda.client;

/**
 * Created by RadiationX on 14.08.2016.
 */
public class OkHttpResponseException extends Exception {
    private int code;
    private String name;
    private String url;

    public OkHttpResponseException(int code, String name, String url) {
        this.code = code;
        this.name = name;
        this.url = url;
    }

    public int getCode() {
        return code;
    }

    @Override
    public String getMessage() {
        return "Response {code=" + code + ", message=" + name + "}";
    }

    public String getName() {
        return name;
    }

    public String getUrl() {
        return url;
    }
}

File: app/src/main/java/forpdateam/ru/forpda/client/OkHttpResponseException.kt
package forpdateam.ru.forpda.client

/**
 * Created by RadiationX on 14.08.2016.
 */
class OkHttpResponseException(val code: Int, val name: String, val url: String) : Exception() {
    override val message: String
        get() = "Response {code=$code, message=$name}"
}
